pytorch_lightning/utilities/jsonargparse_utils.py

```diff
@@ -0,0 +1,150 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from typing import Type
+from jsonargparse import ArgumentParser, ActionConfigFile
+from pytorch_lightning import Trainer, LightningModule, LightningDataModule
+from pytorch_lightning.callbacks import Callback
+
+
+class LightningArgumentParser(ArgumentParser):
+    def __init__(self, *args, **kwargs):
+        """Initialize argument parser that supports configuration file input"""
+        super().__init__(*args, **kwargs)
+        self.add_argument('--cfg',
+            action=ActionConfigFile,
+            help='Path to a configuration file in json or yaml format.')
+
+
+    def add_trainer_args(
+        self,
+        trainer_class: Type[Trainer] = Trainer,
+        nested_key: str = 'trainer'
+    ):
+        """
+        Adds arguments from a trainer class to a nested key of the parser
+
+        Args:
+            trainer_class: Optional extension of the Trainer class.
+            nested_key: Name of the nested namespace where parsed arguments are stored.
+        """
+        assert issubclass(trainer_class, Trainer)
+        self.add_class_arguments(trainer_class, nested_key)
```

Comment:
this mimic out gets default args, right?
does it support also argument types and docs as help?

Reply:
Yes, it gets the arguments with based on its type hints and the description from the docstring goes to the help. The arguments can have complex types and are validated as such, e.g. `Union[int, Dict[int, int]`.